clients/src/test/java/org/apache/kafka/clients/consumer/internals/NetworkClientDelegateTest.java

```diff
@@ -169,14 +183,80 @@ public void testHasAnyPendingRequests() throws Exception {
         }
     }
 
+    @Test
+    public void testPropagateMetadataError() {
+        LinkedList<BackgroundEvent> backgroundEventQueue = new LinkedList<>();
+        Metadata metadata = new Metadata(100, 100, 50000,
```

Comment:
you can do something like this instead of creating a metadata object (which is not needed)

```
        Metadata metadata = mock(Metadata.class);    
        NetworkClientDelegate networkClientDelegate = new NetworkClientDelegate(metadata,
                new BackgroundEventHandler(backgroundEventQueue));

        String exMsg = "Test Auth Exception";
        AuthenticationException authException = new AuthenticationException(exMsg);
        doThrow(authException).when(metadata).maybeThrowAnyException();
```

Reply:
@philipnee Thanks for review!